op-deployer/pkg/deployer/pipeline/opchain.go

```diff
@@ -91,10 +94,88 @@ func DeployOPChain(ctx context.Context, env *Env, bundle ArtifactsBundle, intent
 		DelayedWETHPermissionlessGameProxyAddress: dco.DelayedWETHPermissionlessGameProxy,
 	})
 
+	currentBlock, _ := env.L1Client.BlockNumber(ctx)
+	block, _ := env.L1Client.BlockByNumber(ctx, big.NewInt(int64(currentBlock)))
+	currentBlockHash := block.Hash()
+
+	errCh := make(chan error, 8)
+
+	// If any of the implementations addresses (excluding OpcmProxy) are empty,
+	// we need to set them using the implementation address read from their corresponding proxy.
+	// The reason these might be empty is because we're only invoking DeployOPChain.s.sol as part of the pipeline.
+	// TODO: Need to initialize 'mipsSingletonAddress' and 'preimageOracleSingletonAddress'
+	setImplementationAddressTasks := []func(){
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.DelayedWETHPermissionedGameProxy, currentBlockHash, &st.ImplementationsDeployment.DelayedWETHImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.OptimismPortalProxy, currentBlockHash, &st.ImplementationsDeployment.OptimismPortalImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.SystemConfigProxy, currentBlockHash, &st.ImplementationsDeployment.SystemConfigImplAddress)
+		},
+		func() {
+			setRDPImplementationAddress(ctx, env.L1Client, errCh, dco.AddressManager, &st.ImplementationsDeployment.L1CrossDomainMessengerImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.L1ERC721BridgeProxy, currentBlockHash, &st.ImplementationsDeployment.L1ERC721BridgeImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.L1StandardBridgeProxy, currentBlockHash, &st.ImplementationsDeployment.L1StandardBridgeImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.OptimismMintableERC20FactoryProxy, currentBlockHash, &st.ImplementationsDeployment.OptimismMintableERC20FactoryImplAddress)
+		},
+		func() {
+			setEIP1967ImplementationAddress(ctx, env.L1Client, errCh, dco.DisputeGameFactoryProxy, currentBlockHash, &st.ImplementationsDeployment.DisputeGameFactoryImplAddress)
+		},
+	}
+	for _, task := range setImplementationAddressTasks {
+		go task()
+	}
+
+	var lastTaskErr error
+	for i := 0; i < len(setImplementationAddressTasks); i++ {
+		taskErr := <-errCh
+		if lastTaskErr != nil {
+			lastTaskErr = taskErr
+		}
```

Comment:
I'm not too familiar with channels but this seems like a bug? Since `lastTaskErr` is initialized to `nil` won't we never enter this if block? I think you want

```diff
- if lastTaskErr != nil {
+ if taskErr != nil {
```

Reply:
Ah that does look like a typo/bug.

Reply:
It's meant to be `lastTaskErr == nil` - my bad.